text/0000-deprecate-property-lookup-fallback.md

```diff
@@ -0,0 +1,126 @@
+
+Like [RFC#0276](https://github.com/emberjs/rfcs/blob/68812bf2d439c6bb77ad491e0159b371b68c5c35/text/0276-named-args.md) made argument usage explicit through the `@` prefix, the `this` prefix will resolve the ambiguity and greatly improve clarity, especially in big projects with a lot of files (and uses a lot of addons).
+
+As an aside, the ambiguity that causes confusion for human readers is also a problem for the compiler. While it is not the main goal of this proposal, resolving this ambiguity also helps the rendering system. Currently, the "runtime" template compiler has to perform a helper lookup for every `{{greeting}}` in each template. It will be able to skip this resolution process and perform other optimizations (such as reusing the internal [reference](https://github.com/glimmerjs/glimmer-vm/blob/master/guides/04-references.md)
+object and caches) with this addition.
+
+Furthermore, by enforcing the `this` prefix tooling like the [Ember Language Server](https://github.com/emberwatch/ember-language-server) does not need to know about fallback resolution rules. This makes common features like ["Go To Definition"](https://code.visualstudio.com/docs/editor/editingevolved#_go-to-definition) much easier to implement since we have semantics that mean "property on class".
+
+# Transition Path
+The `{{this.foo}}` syntax has always been supported Handlebars and thus can be used in every version of Ember, however no guides or documentation promote its usage. While applications can start migrating today to this more explicit syntax, we can surface information about what properties on the class were accessed allowing developers to get a concise list of expressions that should be migrated. This can be surfaced because we are already disambiguate the syntax at runtime.
+
+Using this information, we will also have the ability to create a codemod that will help migrate templates.
+
+# How We Teach This
+
+`{{this.foo}}` is the way to access the properties on the component class. This also aligns to property access in JavaScript.
+
+Since the `{{this.foo}}` syntax has worked in Ember.Component (which is the only kind of components available today) since the 1.0 series, we are not really in a rush to migrate the community (and the guides, etc) to using the new syntax. In the meantime, this could be viewed as a tool to improve clarity in templates.
+
+While we think writing {{this.foo}} would be a best practice for new code going forward, the community can migrate at its own pace one component at a time. However, once the fallback functionality is eventually removed this will result in a "Helper not found" error.
+
+## Syntax Breakdown
+The follow is a breakdown of the different forms and what they mean.
+
+- `{{@foo}}` is an argument passed to the component
+- `{{this.foo}}` is a property on the component class
```

Comment:
why is it `this.foo`, but not `this.@foo`?

Reply:
My understanding is that `this.` makes reference to be a property in the component instance, and `@foo`-like arguments do not require a component to even exist, like it will happen in components without a javascript file, that become something similar to functional partials.